provider/manifest/service.go

```diff
@@ -106,9 +106,19 @@ func (s *service) Submit(ctx context.Context, mreq *SubmitRequest) error {
 	case <-ctx.Done():
 		return ctx.Err()
 	case s.mreqch <- req:
-		return <-ch
```

Comment:
@troian  I think this would wait for ever for something to come back on the channel ? I refactored it below. I am unsure if this is the problem you were hitting however.

Reply:
Yes, it would.  Hope this wasn't the problem, but it could be.

The proposed fix would especially help if there was deadline set on the context in the caller (web server).